src/sage/graphs/graph_decompositions/modular_decomposition.pyx

```diff
@@ -1,27 +1,161 @@
+# distutils: language = c++
+# distutils: extra_compile_args = -std=c++11
 r"""
 Modular Decomposition
 
 This module implements the function for computing the modular decomposition
 of undirected graphs.
+
+AUTHORS:
+
+- Lokesh Jain (2017): first implementation of the linear time algorithm of
+  D. Corneil, M. Habib, C. Paul and M. Tedder [TCHP2008]_
+
+- David Einstein (2018): added the algorithm of M. Habib and M. Maurer [HM1979]_
+
+- Cyril Bouvier (2024): second implementation of the linear time algorithm
+  of D. Corneil, M. Habib, C. Paul and M. Tedder [TCHP2008]_
 """
 # ****************************************************************************
 #       Copyright (C) 2017 Lokesh Jain <lokeshj1703@gmail.com>
+#                     2018 David Einstein <deinst@gmail.com>
+#                     2024 Cyril Bouvier <cyril.bouvier@lirmm.fr>
 #
 # This program is free software: you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation, either version 2 of the License, or
 # (at your option) any later version.
 #                  https://www.gnu.org/licenses/
 # ****************************************************************************
+from cython.operator cimport dereference as deref
 
-from enum import Enum, IntEnum
+from enum import IntEnum
 
+from sage.graphs.base.c_graph cimport CGraph, CGraphBackend
+from sage.graphs.graph_decompositions.slice_decomposition cimport \
+        extended_lex_BFS
+from sage.groups.perm_gps.permgroup_element import PermutationGroupElement
 from sage.misc.lazy_import import lazy_import
 from sage.misc.random_testing import random_testing
 
-lazy_import('sage.groups.perm_gps.permgroup_element', 'PermutationGroupElement')
 
+################################################################################
+#                     Corneil-Habib-Paul-Tedder algorithm                      #
+################################################################################
+def corneil_habib_paul_tedder_algorithm(G):
+    r"""
+    Compute the modular decomposition by the algorithm of Corneil, Habib, Paul
+    and Tedder.
+
+    INPUT:
+
+    - ``G`` -- the graph for which modular decomposition tree needs to be
+      computed
+
+    OUTPUT: an object of type Node representing the modular decomposition tree
+    of the graph G
+
+    This function compute the modular decomposition of the given graph by the
```

Comment:
`compute` -> `computes`

Reply:
done in [451b325](https://github.com/sagemath/sage/pull/39038/commits/451b3256ac2cbbb00d2ff24ecc80339fe73d5e40)